File: microlearning-adhd-app/backend/app/schemas/demographics.py
from pydantic import BaseModel


class DemographicsRequest(BaseModel):
    age: int
    gender: str
    highest_education: str
    currently_studying: str
    study_background: str
    adhd_diagnosis: str
    adhd_official_diagnosis: str
    adhd_medication: str
    general_programming_experience: str
    general_programming_languages: str
    general_programming_ability: str
    python_programming_experience: str
    python_programming_ability: str
    device: str


class DemographicsResponse(BaseModel):
    participant_id: str

File: microlearning-adhd-app/frontend/src/services/endpoints/demographics.ts
import api from '../client'

import type { DemographicAnswers } from '../../utils/groupAssignment'
import type { DemographicsSubmission } from '../types/demographics'

export async function postDemographics(participantId: string, demographics: DemographicAnswers) {
  const response = await api.post<DemographicsSubmission>(
    `/participants/${participantId}/demographics`,
    {
      age: Number(demographics.age),
      gender: demographics.gender,
      highest_education: demographics.highestEducation,
      currently_studying: demographics.currentlyStudying,
      study_background: demographics.studyBackground,
      adhd_diagnosis: demographics.adhdDiagnosis,
      adhd_official_diagnosis: demographics.adhdOfficialDiagnosis,
      adhd_medication: demographics.adhdMedication,
      device: demographics.device,
      general_programming_experience: demographics.generalProgrammingExperience,
      general_programming_languages: demographics.generalProgrammingLanguages,
      general_programming_ability: demographics.generalProgrammingAbility,
      python_programming_experience: demographics.pythonProgrammingExperience,
      python_programming_ability: demographics.pythonProgrammingAbility,
    },
  )
  return response.data
}
